sdk/tests/conformance2/rendering/clear-func-buffer-type-match.html

```diff
@@ -0,0 +1,166 @@
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
+
+  debug("");
+  debug("Signed integer buffer");
+
+  var colorbuffer = setupRenderbuffer(gl.COLOR_ATTACHMENT0, gl.RGBA8);
+  var colorbuffer1 = setupRenderbuffer(gl.COLOR_ATTACHMENT1, gl.RGBA32I);
+  gl.drawBuffers([gl.COLOR_ATTACHMENT0, gl.COLOR_ATTACHMENT1]);
+  shouldBe('gl.checkFramebufferStatus(gl.FRAMEBUFFER)', 'gl.FRAMEBUFFER_COMPLETE');
+
+  gl.clear(gl.COLOR_BUFFER_BIT);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "clear and INT buffer");
+
+  gl.clearBufferfv(gl.COLOR, 1, new Float32Array(4));
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "clearBufferfv and INT buffer");
+
+  gl.clearBufferuiv(gl.COLOR, 1, new Uint32Array(4));
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "clearBufferuiv and INT buffer");
+
+  debug("Set up draw buffer so INT buffer is set to NONE");
+  gl.drawBuffers([gl.COLOR_ATTACHMENT0]);
+  gl.clear(gl.COLOR_BUFFER_BIT);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "clear and INT buffer is NONE");
+
+  gl.clearBufferfv(gl.COLOR, 1, new Float32Array(4));
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "clearBufferfv and INT buffer is NONE");
+
+  gl.clearBufferuiv(gl.COLOR, 1, new Uint32Array(4));
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "clearBufferuiv and float buffer");
+
+  debug("");
+  debug("Unsigned integer buffer");
+
+  colorbuffer = setupRenderbuffer(gl.COLOR_ATTACHMENT0, gl.RGBA32UI);
+  colorbuffer1 = setupRenderbuffer(gl.COLOR_ATTACHMENT1, gl.RGBA8);
+  gl.drawBuffers([gl.COLOR_ATTACHMENT0, gl.COLOR_ATTACHMENT1]);
+  shouldBe('gl.checkFramebufferStatus(gl.FRAMEBUFFER)', 'gl.FRAMEBUFFER_COMPLETE');
+
+  gl.clear(gl.COLOR_BUFFER_BIT);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "clear and UINT buffer");
+
+  gl.clearBufferfv(gl.COLOR, 0, new Float32Array(4));
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "clearBufferfv and UINT buffer");
+
+  gl.clearBufferiv(gl.COLOR, 0, new Int32Array(4));
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "clearBufferiv and UINT buffer");
+
+  debug("Set up draw buffer so INT buffer is set to NONE");
+  gl.drawBuffers([gl.NONE, gl.COLOR_ATTACHMENT1]);
+  gl.clear(gl.COLOR_BUFFER_BIT);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "clear and UINT buffer is NONE");
+
+  gl.clearBufferfv(gl.COLOR, 0, new Float32Array(4));
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "clearBufferfv and UINT buffer is NONE");
+
+  gl.clearBufferiv(gl.COLOR, 0, new Int32Array(4));
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "clearBufferiv and float buffer");
+
+  debug("");
+  debug("Float buffer");
+
+  colorbuffer = setupRenderbuffer(gl.COLOR_ATTACHMENT0, gl.RGBA8);
+  var ext = gl.getExtension("EXT_color_buffer_float");
+  if (ext) {
+    debug("EXT_color_buffer_float is available: testing RGBA32F");
+    colorbuffer1 = setupRenderbuffer(gl.COLOR_ATTACHMENT1, gl.RGBA32F);
+  } else {
+    debug("EXT_color_buffer_float is unavailable: testing RGBA8");
+    colorbuffer1 = setupRenderbuffer(gl.COLOR_ATTACHMENT1, gl.RGBA8);
```

Comment:
If RGBA8 is well tested already with these clearBuffer calls, then I would just make this entire test block conditional on the presence of the extension, rather than confusingly running the code with an RGBA8 attachment.